### PR TITLE
Feature/subtopics

### PR DESCRIPTION
**Pull request by MaWoDu**
Introducing subtopics and resources retrieval. 
- All retrievable entities (learning paths, topics, subtpics, resources) have corresponding views which vary in detail level. 
- Started to rethinking the tests structure and purposes to have more focused unit tests.

### CODE REVIEW

File: src/main/java/com/codurance/allaboard/core/actions/topic/FetchTopicById.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class FetchTopicById {
+
+  private final Topics topics;
+
+  @Autowired
+  public FetchTopicById(Topics topics) {
+    this.topics = topics;
+  }
+
+  public Optional<Topic> execute(Long id) {
```
Comment (lagom1985) [inline src/main/java/com/codurance/allaboard/core/actions/topic/FetchTopicById.java:19]: I know why you use execute but we didn't use it anywhere also so think carefully if applying it or not

File: src/main/java/com/codurance/allaboard/web/controllers/learningpath/LearningPathController.java [hunk 2]
```diff
@@ -55,16 +55,14 @@ public LearningPathController(
...
       logger.info("Learning path with id: [{}] not found", id.toString());
       return new ResponseEntity<>(HttpStatus.NOT_FOUND);
     }
-    LearningPath learningPath = optional.get();
-    LearningPathDetailView learningPathDetailView = new LearningPathDetailView(learningPath.getId(),
-        learningPath.getName(),
-        learningPath.getDescription(),
-        learningPath.getTopics());
+
+    LearningPathDetailView learningPathDetailView = LearningPathDetailView.from(learningPath.get());
```
Comment (lagom1985) [inline src/main/java/com/codurance/allaboard/web/controllers/learningpath/LearningPathController.java:65]: really nice guys ! 👍🏻

**Review by lagom1985 — APPROVED** [review]
Good job! The only thing I don't agree with is the creation of a e2e package for the utils. In my opinion they belong to the web package like I originally did. Cheers

File: src/main/java/com/codurance/allaboard/core/actions/topic/FetchTopicById.java (hunk 1 above)
Comment (MaWoDu) [reply src/main/java/com/codurance/allaboard/core/actions/topic/FetchTopicById.java:19]: Generally, the idea is to unify the interface underlying all actions and this is the first step towards it. 
Also, the tautology the original code presents, makes the actual code usage difficult, for example, saveSurvey.save(survey);

**Pull request merged**